### PR TITLE
Use httpx for links instead of requests

### PR DESCRIPTION
**Pull request by Harry-Jing**
I use latest version to modify

### CODE REVIEW

**Review by acheong08 — APPROVED** [review]
Looks good. Will test it out

**Comment by Harry-Jing** [comment]
The code I previous pull works better on my computer
but I'm not sure with this

**Comment by Harry-Jing** [comment]
Sometimes httpx need to specifically set the timeout, it's easy to have timeouterror i default

**Comment by acheong08** [comment]
@Harry-Jing post() 
`got an unexpected keyword argument 'stream'`

How do you stream with httpx?

**Comment by Harry-Jing** [comment]
https://www.python-httpx.org/async/#streaming-responses

**Comment by Harry-Jing** [comment]
sorry, I forget to modify this

**Comment by acheong08** [comment]
Is it possible to do this without making the function async?

**Comment by Harry-Jing** [comment]
ture

**Comment by Harry-Jing** [comment]
https://www.python-httpx.org/compatibility/#streaming-responses

**Comment by Harry-Jing** [comment]
image

**Comment by acheong08** [comment]
Ah nvm then. Let me try again with that

**Comment by Harry-Jing** [comment]
Sorry to trouble you

**Comment by acheong08** [comment]
@Harry-Jing 
```
        response = httpx.stream("POST",
                                "https://chat.openai.com/backend-api/conversation",
                                headers=self.headers,
                                data=json.dumps(data),
                                timeout=50,
                                )
        for line in response.iter_lines():
```
response doesn't have iter lines function even though it is documented

**Comment by acheong08** [comment]
More errors with index out of range

**Comment by acheong08** [comment]
@Harry-Jing Could you try to implement this with httpx and make a new pull request. There are some urgent errors I need to fix that can't be held up by this

**Comment by Harry-Jing** [comment]
no problem

**Comment by Harry-Jing** [comment]
image

**Comment by Harry-Jing** [comment]
is this run normaly?

**Comment by acheong08** [comment]
since it's a stream, you have to loop through it.
```py
for msg in resp:
	...
```

**Comment by Harry-Jing** [comment]
I have already try this, but nothing show
and now i know reason
image

**Comment by Harry-Jing** [comment]
So I have question to this

**Comment by Harry-Jing** [comment]
image
It's very strange that this str cant convert into dict

**Comment by Harry-Jing** [comment]
Now i find is becase httpx will have \n suffix compare to requests

**Comment by Harry-Jing** [comment]
image
now it seem work proberly
